### PR TITLE
feat: clean finished job experiments on clean finished resources flow

### PR DESCRIPTION
**Pull request by GuyAb**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔/✖                                                                        |
| New feature     | ✔/✖                                                                        |
| Breaking change | ✔/✖                                                                        |
| Deprecations    | ✔/✖                                                                        |
| Documentation   | ✔/✖                                                                        |
| Tests added     | ✔/✖                                                                        |

Related issues: #XXX , #XXX ...
Closes #XXX , #XXX ...

Further  information:

### CODE REVIEW

File: src/jobs/models/kubernetes/jobExperimentsHandler.js [hunk 1]
```diff
@@ -58,6 +58,10 @@ async function stopChaosExperimentsForJob(jobId){
     await chaosExperimentsManager.stopJobExperimentsByJobId(jobId);
 }
 
+async function clearAllFinishedJobExperiments(){
+    return await chaosExperimentsManager.clearAllFinishedResources();
```
Comment (kerenfi) [inline src/jobs/models/kubernetes/jobExperimentsHandler.js:62]: it a one liner - why not calling it from main function of `deleteAllContainers`

File: src/jobs/models/kubernetes/jobExperimentsHandler.js [hunk 2]
```diff
@@ -77,5 +81,6 @@ function convertDurationStringToMillisecond(durationString) {
 module.exports = {
     setChaosExperimentsIfExist,
     scheduleChaosExperiment,
-    stopChaosExperimentsForJob
+    stopChaosExperimentsForJob,
+    clearAllFinishedJobExperiments
```
Comment (kerenfi) [inline src/jobs/models/kubernetes/jobExperimentsHandler.js:85]: why is it exposed externally - is it only to allow unit tests stubs

File: src/jobs/models/kubernetes/jobConnector.js [hunk 3]
```diff
@@ -84,7 +84,8 @@ module.exports.deleteAllContainers = async (jobPlatformName) => {
             deleted++;
         }
     }
-    return { deleted };
+    const jobExperimentsDeleted = jobExperimentHandler.clearAllFinishedJobExperiments();
+    return { deleted, internal_resources_deleted: jobExperimentsDeleted };
```
Comment (kerenfi) [inline src/jobs/models/kubernetes/jobConnector.js:88]: internal_resources_deleted:  I think we should open it to expansion and have it as an object (maybe in the future we'll have other internal resources)

like
```
internal_resources_deleted: {
 chaos-mesh: x
}
```

File: src/jobs/models/kubernetes/jobExperimentsHandler.js (hunk 2 above)
Comment (GuyAb) [reply src/jobs/models/kubernetes/jobExperimentsHandler.js:85]: no, it is called by k8s jobConnector

File: src/jobs/models/kubernetes/jobConnector.js (hunk 3 above)
Comment (GuyAb) [reply src/jobs/models/kubernetes/jobConnector.js]: done.

**Pull request merged**